### PR TITLE
📖 Update concepts doc

### PR DESCRIPTION
**Pull request by mboersma**
**What this PR does / why we need it**:
Simplifies and hopefully clarifies some of the content in the Concepts doc.

I had to make changes while trying to grok this content, so I apologize in advance if any of this represents my misunderstanding or is unhelpful.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/concepts.md [hunk 1]
```diff
@@ -5,73 +5,74 @@
...
 ## Custom Resource Definitions (CRDs)
 
 ### Machine
 
-A "Machine" is the declarative spec for a Node, as represented in Kubernetes core. If a new Machine object is created, a provider-specific controller will handle provisioning and installing a new host to register as a new Node matching the Machine spec. If the Machine's spec is updated, a provider-specific controller is responsible for updating the Node in-place or replacing the host with a new one matching the updated spec. If a Machine object is deleted, the corresponding Node should have its external resources released by the provider-specific controller, and should be deleted as well.
+A "Machine" is the declarative spec for a Node, as represented in Kubernetes core. If a new Machine object is created, a provider-specific controller will provision and install a new host to register as a new Node matching the Machine spec. If the Machine's spec is updated, the controller updates the Node in-place, or replaces the host with a new one matching the updated spec. If a Machine object is deleted, its host resources and corresponding Node will be deleted by the controller.
```
Comment (fabriziopandini) [inline docs/book/src/user/concepts.md:44]: > A "Machine" is the declarative spec for a Node

I think it is more appropriate something like:
A "Machine" is the declarative spec for an infrastructure component hosting a Kubernetes node. e.g  a VM

> the controller updates the Node in-place

AFAIK this is not true

> its host resources

It's underlying infrastructure

File: docs/book/src/user/concepts.md [hunk 2]
```diff
@@ -5,73 +5,74 @@
...
 
-Fields like the kubelet version are modeled as fields on the Machine's spec. Any other information that is provider-specific, though, is part of the InfraProviderRef and is not portable between different providers.
+Common fields such as kubelet version are modeled as fields on the Machine's spec. Any information that is provider-specific is part of the InfraProviderRef and is not portable between different providers.
```
Comment (fabriziopandini) [inline docs/book/src/user/concepts.md:46]:
```suggestion
Common fields such as Kubernetes version are modeled as fields on the Machine's spec. Any information that is provider-specific is part of the InfraProviderRef and is not portable between different providers.
```

File: docs/book/src/user/concepts.md [hunk 3]
```diff
@@ -5,73 +5,74 @@
...
 
-A cluster whose lifecycle is managed by the Management cluster.
+A Kubernetes cluster whose lifecycle is managed by a management cluster.
 
 ### Infrastructure provider
 
-A source of computational resources (e.g. machines, networking, etc.). Examples for cloud include AWS, Azure, Google, etc.; for bare metal include VMware, MAAS, metal3.io, etc. When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS) each way is referred to as a variant.
+A source of computational resources, such as machines and networking. For example, cloud infrastructure providers incluede AWS, Azure, and Google, and bare metal infrastructure providers include VMware, MAAS, and metal3.io.
+
+When there is more than one way to obtain resources from the same infrastructure provider--such as AWS offering both EC2 and EKS--each way is referred to as a variant.
```
[the remark this answers is not in the document]
Comment (mboersma) [reply docs/book/src/user/concepts.md:18]: I was going for an em-dash, but strictly speaking that only renders correctly in Markdown using the HTML entity `&mdash;`. That's very hard to read in plain text. I'll use parentheses instead.

File: docs/book/src/user/concepts.md [hunk 4]
```diff
@@ -5,73 +5,74 @@
 
 ### Management cluster
 
-The cluster where one or more Infrastructure Providers run, and where resources (e.g. Machines) are stored.  Typically referred to when you are provisioning multiple clusters.
+A Kubernetes cluster that manages the lifecycle of workload clusters. A management cluster is also where one or more infrastructure providers run, and where resources such as machines are stored.
```
[the remark this answers is not in the document]
Comment (mboersma) [reply docs/book/src/user/concepts.md:8]: I think you're right, but the capitalization was inconsistently used in the first section, so I guess I got rid of it. Instead I'll restore it and try to make it more consistent.

**Comment by mboersma** [comment]
>  the lines are long in this doc

I agree. Markdown (and similar markup languages) should be readable as plain text, which implies having a consistent right margin.

But that makes the initial diff with the reformatting a pain to review, and subsequent docs PRs will also generate more trivial diffs just from reflowing the lines. So I follow local custom, which here seemed to be "don't bother with line breaks." I'm happy to reformat it though.

**Comment by JoelSpeed** [comment]
> But that makes the initial diff with the reformatting a pain to review, and subsequent docs PRs will also generate more trivial diffs just from reflowing the lines. So I follow local custom, which here seemed to be "don't bother with line breaks." I'm happy to reformat it though.

This seems like reasonable logic not to change, I don't have a strong opinion either way, happy to leave as is now :)

File: docs/book/src/user/concepts.md [hunk 5]
```diff
@@ -5,73 +5,74 @@
...
-### Workload/Target Cluster
+### Workload cluster
 
-A cluster whose lifecycle is managed by the Management cluster.
+A Kubernetes cluster whose lifecycle is managed by a Management Cluster.
 
 ### Infrastructure provider
 
-A source of computational resources (e.g. machines, networking, etc.). Examples for cloud include AWS, Azure, Google, etc.; for bare metal include VMware, MAAS, metal3.io, etc. When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS) each way is referred to as a variant.
+A source of computational resources, such as Machines and networking. For example, cloud Infrastructure Providers incluede AWS, Azure, and Google, and bare metal Infrastructure Providers include VMware, MAAS, and metal3.io.
```
Comment (JoelSpeed) [inline docs/book/src/user/concepts.md:16]: Typo
```suggestion
A source of computational resources, such as Machines and networking. For example, cloud Infrastructure Providers include AWS, Azure, and Google, and bare metal Infrastructure Providers include VMware, MAAS, and metal3.io.
```

File: docs/book/src/user/concepts.md [hunk 6]
```diff
@@ -5,73 +5,74 @@
...
+* __Pod-based__ deployments require an external hosting cluster. The control plane components are deployed using standard *Deployment* and *StatefulSet* objects and the API is exposed using a *Service*.
 
-* __Pod Based__  deployments require an external hosting cluster, the control plane is deployed using standard *Deployment* and *StatefulSet* objects and then the API exposed using a *Service*.
+* __External__ control planes are offered and controlled by some system other than Cluster API, such as GKE, AKS, EKS, or IKS.
 
-* __External__ control planes are offered and controlled by some system other than Cluster API (e.g., GKE, AKS, EKS, IKS).
+As of v1alpha2, __machine-based__ is the only control plane type that Cluster API supports.
```
Comment (JoelSpeed) [inline docs/book/src/user/concepts.md:38]: Should this capitalisation match the title above? 
```suggestion
As of v1alpha2, __Machine-Based__ is the only control plane type that Cluster API supports.
```

File: docs/book/src/user/concepts.md [hunk 7]
```diff
@@ -5,73 +5,74 @@
...
 
-Remediation of Nodes is performed by deleting the Machine that created the Node.
-MachineHealthChecks will only remediate Nodes if they are owned by a MachineSet,
-this ensures that the Kubernetes cluster does not lose capacity, as the MachineSet will create a new Machine to replace the failed Machine.
+MachineHealthChecks will only remediate Nodes if they are owned by a MachineSet. This ensures that the Kubernetes cluster does not lose capacity, since the MachineSet will create a new Machine to replace the failed Machine.
 
 ### BootstrapData
 
-BootstrapData contains the machine or node role specific initialization data (usually cloud-init) used by the infrastructure provider to bootstrap a machine into a node.
+BootstrapData contains the Machine- or Node role-specific initialization data (usually cloud-init) used by the Infrastructure Provider to bootstrap a Machine into a Node.
```
Comment (JoelSpeed) [inline docs/book/src/user/concepts.md:76]: Unintentional hash?
```suggestion
BootstrapData contains the Machine or Node role-specific initialization data (usually cloud-init) used by the Infrastructure Provider to bootstrap a Machine into a Node.
```

File: docs/book/src/user/concepts.md (hunk 6 above)
Comment (detiber) [reply docs/book/src/user/concepts.md:38]: I'm wondering if we should update this language to indicate that other control plane types are now possible through the use of "Control Plane providers"?

**Comment by sedefsavas** [comment]
Failed due to quay outage.

/test pull-cluster-api-e2e

**Comment by fabriziopandini** [comment]
@mboersma please squash, then lgtm

**Pull request merged**